[CI] Added GitHub CI to LibSDCParse

AlexandreSinger opened the pull request:
This PR adds a test workflow which builds LibSDCParse's tester executable and runs a suite of SDC files to parse.

Also removed the old Travis yaml file.

Resolves #4

AlexandreSinger committed:
[CI] Added GitHub CI to LibSDCParse

This PR adds a test workflow which builds LibSDCParse's tester
executable and runs a suite of SDC files to parse.

Also removed the old Travis yaml file.

AmirhosseinPoolad commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,35 @@
+  pull_request:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * *' # daily
```
Do we want to run this daily? Is there any point?

AlexandreSinger replied:
Fair point. I was originally on the fence about this. The CI is quick, so its not that harmful; but I do not think there is enough reason to have it. I will remove it.

AmirhosseinPoolad reviewed (commented):
Sorry for the late review! Looks good, had a small comment/question.

AlexandreSinger committed:
[CI] Removed Daily Test Run

There is no clear reason to make the CI run daily. We can always add it
in the future if it is found to be useful.
